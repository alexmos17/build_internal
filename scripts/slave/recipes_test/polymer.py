# Copyright (c) 2013 The Chromium Authors. All rights reserved.
# Use of this source code is governed by a BSD-style license that can be
# found in the LICENSE file.

"""Test inputs for recipes/polymer.py"""

def basic_test(api):
  return {
    'build_properties': {
      'repository': 'https://github.com/Polymer/polymer'
    }
  }
