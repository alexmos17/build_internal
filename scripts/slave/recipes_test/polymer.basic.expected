[
  {
   "cmd": ["[BUILD_ROOT]/scripts/slave/git_setup.py", "--path", "[SLAVE_BUILD_ROOT]/polymer", "--url", "https://github.com/Polymer/polymer.git"],
   "name": "git setup",
   "seed_steps": ["git setup", "git fetch", "git update-ref", "git clean", "git checkout", "git submodule"]
  },
  {
   "cmd": ["git", "fetch", "origin", "--recurse-submodules"],
   "cwd": "[SLAVE_BUILD_ROOT]/polymer",
   "name": "git fetch"
  },
  {
   "cmd": ["git", "update-ref", "refs/heads/master", "origin/master"],
   "cwd": "[SLAVE_BUILD_ROOT]/polymer",
   "name": "git update-ref"
  },
  {
   "cmd": ["git", "clean", "-f", "-d", "-x"],
   "cwd": "[SLAVE_BUILD_ROOT]/polymer",
   "name": "git clean"
  },
  {
   "cmd": ["git", "checkout", "-f", "master"],
   "cwd": "[SLAVE_BUILD_ROOT]/polymer",
   "name": "git checkout"
  },
  {
   "cmd": ["git", "submodule", "update", "--init", "--recursive"],
   "cwd": "[SLAVE_BUILD_ROOT]/polymer",
   "name": "git submodule"
  },
  {
   "cmd": ["mkdir", "[SLAVE_BUILD_ROOT]/polymer/.tmp"],
   "name": "mktmp"
  },
  {
   "cmd": ["npm", "install", "--tmp", "[SLAVE_BUILD_ROOT]/polymer/.tmp"],
   "cwd": "[SLAVE_BUILD_ROOT]/polymer",
   "name": "update-install"
  },
  {
   "allow_subannotations": true,
   "cmd": ["xvfb-run", "grunt", "test-buildbot"],
   "cwd": "[SLAVE_BUILD_ROOT]/polymer",
   "name": "test"
  }
]